### PR TITLE
Convert common helpers to Typescript

### PR DESCRIPTION
**Pull request by daniellesniak**
See flarum/framework#3533

### CODE REVIEW

File: js/src/common/helpers/avatar.tsx [hunk 1]
```diff
@@ -1,26 +1,28 @@
+import * as Mithril from 'mithril';
+import User from '../models/User';
+
 /**
  * The `avatar` helper displays a user's avatar.
  *
- * @param {User} user
- * @param {Object} attrs Attributes to apply to the avatar element
- * @return {Object}
+ * @param attrs Attributes to apply to the avatar element
```
Comment (askvortsov1) [inline js/src/common/helpers/avatar.tsx:7]: I wouldn't want to have one param documented but the other not

File: js/src/common/helpers/avatar.tsx [hunk 2]
```diff
@@ -1,26 +1,28 @@
...
  */
-export default function avatar(user, attrs = {}) {
+
```
Comment (askvortsov1) [inline js/src/common/helpers/avatar.tsx:9]: Extra space?

**Review by askvortsov1 — CHANGES_REQUESTED** [review]
Looking good, just a few nitpicks

File: js/src/common/helpers/avatar.tsx (hunk 1 above)
Comment (datitisev) [reply js/src/common/helpers/avatar.tsx:7]: The other is still documented in the typehinting. Attrs only remains for the description in the JSDoc. I don't know if we want to do that or keep tham all, howeverr 🤷

Comment (daniellesniak) [reply js/src/common/helpers/avatar.tsx:7]: I think we should leave all parameters in JSDoc if there's any with description and get rid of all parameters if none of them have any description. By the way this should resolve PHPStorm's warning "Parameter is not described in JSDoc".

File: js/src/common/helpers/avatar.tsx (hunk 2 above)
Comment (daniellesniak) [reply js/src/common/helpers/avatar.tsx]: Yes, fixed in avatar.tsx and username.tsx.

**Review by askvortsov1 — APPROVED** [review]
Thank you!

**Review by dsevillamartin — APPROVED** [review]
:+1:

**Pull request merged**